pkg/apis/core/validation/validation.go

```diff
@@ -5642,31 +5672,45 @@ func ValidateService(service *core.Service) field.ErrorList {
 		ports[key] = true
 	}
 
-	// Validate SourceRanges field or annotation.
+	// Validate SourceRanges field or annotation. Existing invalid CIDR values do not
+	// need to be fixed. Note that even with the tighter CIDR validation we still
+	// allow excess whitespace, because that is effectively part of the API, and there
+	// are no security-related arguments for fixing it.
 	if len(service.Spec.LoadBalancerSourceRanges) > 0 {
 		fieldPath := specPath.Child("LoadBalancerSourceRanges")
 
 		if service.Spec.Type != core.ServiceTypeLoadBalancer {
 			allErrs = append(allErrs, field.Forbidden(fieldPath, "may only be used when `type` is 'LoadBalancer'"))
 		}
+		var existingSourceRanges sets.Set[string]
```

Comment:
Should we capture this pattern in something like `ratchetValidate[T comparable](existing []T,  next func() (T, bool), validate func(T) ErrorList)` ?  Even just local to this file?

Reply:
We have not tried to tackle ratcheting in declarative validation yet, but this might be the first use-case to explore it.

Reply:
I'm not sure what you mean there (particularly `next`). But the different sections are all just different enough that it seemed like trying to add an abstraction would end up resulting in _more_ code...